common/v2/features/SendAssets/stateFactory.tsx

```diff
@@ -86,18 +89,22 @@ const TxConfigFactory: TUseApiFactory<State> = ({ state, setState }) => {
 
     provider
       .sendRawTx(signedTx)
-      .then(transactionReceipt => {
+      .then(retrievedTransactionReceipt => {
+        const transactionReceipt = fromTxReceiptObj(retrievedTransactionReceipt);
+        addNewTransactionToAccount(state.txConfig.senderAccount, transactionReceipt || {});
         setState((prevState: State) => ({
           ...prevState,
-          txReceipt: fromTxReceiptObj(transactionReceipt)
+          txReceipt: transactionReceipt
         }));
       })
       .catch(txHash => {
         // If rejected, data is a tx hash, not a receipt. Fetch the receipt, then save receipt for flow
-        provider.getTransactionByHash(txHash).then(transactionReceipt => {
+        provider.getTransactionByHash(txHash).then(retrievedTransactionReceipt => {
```

Comment:
By reducing indentation levels and nesting we can increase readability.
Since `then()` will wait for the resolve of the previous return value we can chain Promises,
and you can continue chaining even after `catch`:
```
.catch(txHash => provider.getTransaction(txHash))
.then(retrievedTransactionReceipt => ...)
```

Reply:
It's strange, the code still shows: 
<img width="788" alt="Screenshot 2019-09-02 at 15 14 12" src="https://user-images.githubusercontent.com/2429708/64113937-58ca0800-cd94-11e9-9ea4-f7351bd807a5.png">

When the goal is to write:
```
provider.sendRawTx(signedTx)
 .then(retrievedTxReceipt => retrievedTransactionReceipt)
 .catch(txHash => provider.getTransactionByHash(txHash))
 .finally(retrievedTxReceipt => {
    const txReceipt  = fromTxReceiptObj(retrievedTransactionReceipt);
    addNewTransactionToAccount(state.txConfig.senderAccount, txReceipt || {});
    setState((prevState: State) => ({
      ...prevState,
       txReceipt
    }));
 })
```
Am I missing something?

Reply:
> .finally(retrievedTxReceipt => { ...
This doesn't work with TS.

This has error: `Argument of type '(retrievedTxReceipt: ITxReceipt) => void' is not assignable to parameter of type '() => void'.`